File: 2275-largest-combination-with-bitwise-and-greater-than-zero/2275-largest-combination-with-bitwise-and-greater-than-zero.py
from typing import List

class Solution:
    def largestCombination(self, candidates: List[int]) -> int:
        max_combination_size = 0
        
        # Loop over each bit position (0 to 23 for numbers up to 10^7)
        for bit in range(24):
            count = 0
            
            # Count how many numbers have the current bit set
            for num in candidates:
                if num & (1 << bit):
                    count += 1
            
            # Update the maximum combination size with the current count
            max_combination_size = max(max_combination_size, count)
        
        return max_combination_size

File: 2275-largest-combination-with-bitwise-and-greater-than-zero/2275-largest-combination-with-bitwise-and-greater-than-zero.cpp
class Solution {
public:
    int largestCombination(vector<int>& candidates) {
        int maxCombinationSize = 0;
        
        // Iterate through each bit position from 0 to 23 (since max(candidates[i]) <= 10^7)
        for (int bit = 0; bit < 24; ++bit) {
            int count = 0;
            
            // Count how many numbers have the current bit set
            for (int num : candidates) {
                if (num & (1 << bit)) {
                    count++;
                }
            }
            
            // Update the maximum size of a valid combination
            maxCombinationSize = max(maxCombinationSize, count);
        }
        
        return maxCombinationSize;
    }
};